great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -29,11 +29,16 @@
     Datasource,
 )
 
+try:
+    from sqlalchemy import create_engine
+    from sqlalchemy.engine import Engine
+except ImportError:
+    create_engine = None
+    Engine = None
```

Comment:
I'd like it if we imported the top-level packages so that it's more apparent in the annotations that `Engine` is a `sqlalchemy.engine.Engine`?
```suggestion
SQLA_IMPORTED: bool = False
try:
    import sqlalchemy
    SQLA_IMPORTED: bool = True
except ImportError:
    pass
```